### ISSUE TITLE
Fix S3353 FP: With Tuple Assignment

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#3271]
Side effect of #3239, rule S3353 is also raised

```C#
var tmp = 0;
var flag = true;
(flag, tmp) = (false, 5);
```

**Comment by andrei-epure-sonarsource** [comment]
is there a way to figure out what other rules could have been affected by this?

**Comment by pavel-mikula-sonarsource** [comment]
AFAIK there's only peach and the hard way (manually try it rule by rule).

**Comment by pavel-mikula-sonarsource** [comment]
Another example reported in #3661: S3353 raises a false positive with implicit conversion from a constant value.

### Repro steps

```C#
public struct MyStruct<T>
{
    public static implicit operator MyStruct<T>(T value) { return default; }
}

public static class Example
{
    public static MyStruct<int> Get()
    {
        MyStruct<int> x = 1;
        return x;
    }
}
```

We should check if the type can be declared with `const` modifier.
